### PR TITLE
Change the table view to remove collapsed list items

### LINKED ISSUE #1227: TreeTable animation stutters when collapsing a large subtree.
Collapse performance is bad when a large block is collapsed.
The problem is in this case you will have 100s of rows briefly on the screen at once when each row is very short. To repro, use a 15 second profile sample to create a large tree.
A reasonable optimization is to ensure that we don't animate rows that were not visible at the start of the animation. There is no need to include those rows in the animation as the user was never aware they were in the subtree that was collapsed.
collapse_example

### PR DESCRIPTION
**Pull request by DaveShuckerow**
This avoids building 0-height widgets, which was the cause of the terrible lag we were seeing.

Fixes #1227

### CODE REVIEW

File: packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart [hunk 1]
```diff
@@ -22,42 +23,90 @@ class CpuProfiler extends StatelessWidget {
 
   final Function(CpuStackFrame stackFrame) onStackFrameSelected;
 
+  static const Key expandButton = Key('CpuProfiler - Expand Button');
+  static const Key collapseButton = Key('CpuProfiler - Collapse Button');
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: nit: r/expandButton/expandButtonKey ? this seems to be the naming pattern we use elsewhere

File: packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart [hunk 2]
```diff
@@ -22,42 +23,90 @@ class CpuProfiler extends StatelessWidget {
...
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            TabBar(
+              labelColor: Theme.of(context).textTheme.body1.color,
+              isScrollable: true,
+              controller: _tabController,
+              tabs: CpuProfiler.tabs,
+            ),
+            if (_tabController.index != 0)
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: The order of the tabs is different on the timeline and the performance page. This should instead check that the selected tab is not the flame chart tab.

File: packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart [hunk 3]
```diff
@@ -80,23 +129,19 @@ class CpuProfiler extends StatelessWidget {
...
         width: constraints.maxWidth * 2,
-        selected: selectedStackFrame,
-        onSelected: (sf) => onStackFrameSelected(sf),
+        selected: widget.selectedStackFrame,
+        onSelected: (sf) => widget.onStackFrameSelected(sf),
       );
     });
 
     // TODO(kenz): tree table is extremely slow with large data set. It should
     // be optimized before including in the profiler.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: if this is fixed, please remove the todo

Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: Done.

File: packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart (hunk 2 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: Done, using a key.

Note that the current implementation is using the same tab list on both pages.  If we want a different order, we'll need to provide an alternate tabs list.

File: packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart (hunk 1 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart]: Done.

**Pull request merged**